elastic/apm-server
# aggregation/txmetrics: set timeseries.instance

## Motivation/summary

Set an ID for each unique transaction aggregation key, using the existing `timeseries.instance` field defined in libbeat.

We incorporate the full service and transaction names to offset the risk of hash collisions.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

1. Enable aggregation
2. Send some transactions
3. Check that the metric docs have a `timeseries.instance` field, which includes the service name, transaction name, and a hash

## Related issues

Closes #3813
